src/core/query.ts

```diff
@@ -112,7 +112,11 @@ export class Query {
   }
 
   withoutSorts(...sorts: Sort[]): Query {
-    this.request.sort = this.request.sort.filter((oldSort) => !sorts.find((sort) => sort.field === oldSort.field));
+    this.request.sort = this.request.sort.filter((oldSort) => !sorts.find((sort) =>
+      sort.type === 'ByIds'
+        ? oldSort.type === 'ByIds' && sort.ids.toString() === oldSort.ids.toString()
+        : sort.field === oldSort['field']
```

Comment:
I think you should cast `oldSort`. Using `[]` to get around types should feel like a hack.

Reply:
fixed